tests/Export/ProductServiceTest.php

```diff
@@ -237,4 +238,97 @@ public function testVariantsWithSameParentButDifferentDisplayGroupAreExportedAsS
         $this->assertSame($expectedFirstVariantId, $products[0]->getId());
         $this->assertSame($expectedSecondVariantId, $products[1]->getId());
     }
+
+    public function testTheMainProductIsBasedOnTheMainVariantIdOfADisplayGroup(): void
+    {
+        $expectedParentId = Uuid::randomHex();
+        $expectedFirstVariantId = Uuid::randomHex();
+        $expectedSecondVariantId = Uuid::randomHex();
+        $expectedMainVariantId = $expectedSecondVariantId;
+
+        $firstOptionId = Uuid::randomHex();
+        $secondOptionId = Uuid::randomHex();
+        $optionGroupId = Uuid::randomHex();
+
+        $this->createVisibleTestProduct([
+            'id' => $expectedParentId,
+            'active' => false,
+            'configuratorSettings' => [
+                [
+                    'option' => [
+                        'id' => $firstOptionId,
+                        'name' => 'Red',
+                        'group' => [
+                            'id' => $optionGroupId,
+                            'name' => 'Color',
+                        ],
+                    ],
+                ],
+                [
+                    'option' => [
+                        'id' => $secondOptionId,
+                        'name' => 'Orange',
+                        'group' => [
+                            'id' => $optionGroupId,
+                            'name' => 'Color',
+                        ],
+                    ],
+                ],
+            ],
+            'configuratorGroupConfig' => [
+                [
+                    'id' => $optionGroupId,
+                    // Explicitly set this to false. This tells Shopware to consider the mainVariationId (if set).
+                    'expressionForListings' => false,
+                    'representation' => 'box'
+                ]
+            ],
+        ]);
+
+        $this->createVisibleTestProduct($this->getBasicVariantData([
+            'id' => $expectedFirstVariantId,
+            'parentId' => $expectedParentId,
+            'productNumber' => 'FINDOLOGIC001.1',
+            'name' => 'FINDOLOGIC VARIANT 1',
+            'options' => [
+                ['id' => $firstOptionId]
+            ],
+        ]));
+
+        $this->createVisibleTestProduct($this->getBasicVariantData([
+            'id' => $expectedSecondVariantId,
+            'parentId' => $expectedParentId,
+            'productNumber' => 'FINDOLOGIC001.2',
+            'name' => 'FINDOLOGIC VARIANT 2',
+            'options' => [
+                ['id' => $secondOptionId]
+            ],
+        ]));
+
+        $this->getContainer()->get('product.repository')->update([
+            [
+                'id' => $expectedFirstVariantId,
+                'mainVariantId' => $expectedMainVariantId
```

Comment:
It's not possible to set the mainVariationId directly at createVisibleTestProduct?

Reply:
No because at this point the related product does not exist yet, which causes an SQL constraint violation.